packages/plans-grid-next/src/components/actions.tsx

```diff
@@ -454,59 +354,69 @@ const PlanFeaturesActionsButton: React.FC< PlanFeaturesActionsButtonProps > = (
 		storageAddOnsForPlan,
 	} );
 
-	const onCtaClick = useActionCallback( {
+	const priceString = formatCurrency(
+		( discountedPrice.monthly || originalPrice.monthly ) ?? 0,
+		currencyCode || 'USD',
+		{
+			stripZeros: true,
+			isSmallestUnit: true,
+		}
+	);
+
+	const { callback, text, postButtonText, status } = useAction( {
+		// TODO: Double check that we need to do this boolean coercion
+		availableForPurchase,
+		isLargeCurrency: !! isLargeCurrency,
+		isStuck,
 		planSlug,
+		planTitle,
+		priceString,
 		cartItemForPlan,
 		selectedStorageAddOn,
 	} );
 
-	// TODO: Unsure about using free plan as a fallback. We should revisit.
-	const onFreeTrialCtaClick = useActionCallback( {
+	const { callback: freeTrialCallback, text: freeTrialText } = useAction( {
+		// TODO: Double check that we need to do this boolean coercion
+		isFreeTrialAction: true,
+		isLargeCurrency: !! isLargeCurrency,
+		isStuck,
+		// TODO: Unsure about using free plan as a fallback. We should revisit.
 		planSlug: freeTrialPlanSlug ?? PLAN_FREE,
+		planTitle,
+		priceString,
 		cartItemForPlan: { product_slug: freeTrialPlanSlug ?? PLAN_FREE },
+		selectedStorageAddOn,
 	} );
 
 	if ( isWpcomEnterpriseGridPlan( planSlug ) ) {
 		return (
-			<PlanButton planSlug={ planSlug } onClick={ onCtaClick }>
+			<PlanButton planSlug={ planSlug } onClick={ callback }>
 				{ translate( 'Learn more' ) }
 			</PlanButton>
 		);
 	}
 
-	const priceString = formatCurrency(
-		( discountedPrice.monthly || originalPrice.monthly ) ?? 0,
-		currencyCode || 'USD',
-		{
-			stripZeros: true,
-			isSmallestUnit: true,
-		}
-	);
-
 	if ( isLaunchPage ) {
 		return (
 			<LaunchPagePlanFeatureActionButton
 				planSlug={ planSlug }
-				planTitle={ planTitle }
-				priceString={ priceString }
-				isStuck={ isStuck }
-				isLargeCurrency={ !! isLargeCurrency }
-				onCtaClick={ onCtaClick }
+				onCtaClick={ callback }
+				text={ text }
 			/>
 		);
 	}
 	if ( isInSignup ) {
```

Comment:
I wonder if it might help to start by merging the two into a single function: `SignupFlowPlanFeatureActionButton` and `LoggedInPlansFeatureActionButton`. Ultimately, we don't need a factory method of sorts here and the end goal ought to be an extremely generic method that renders whatever is passed through the `useAction` hook (give other things that need to be done here / price-string, selected-storage, etc.).

Reply:
Agree here. My one reservation is making too many large, sweeping changes at once.

Similarly to how we split up [refactoring the useActionCallback]( https://github.com/Automattic/wp-calypso/pull/88694 ) from the work we're doing in this existing PR, how would we feel about merging the buttons in one more smaller, focused follow-up PR @chriskmnds?

Reply:
It's ok to do it in pieces, certainly. It needs a brave attack to reap everything out, including `isInSignup` etc... we are heading there, so at any point we feel confident there is something in stare of merging, let's merge and continue. I think removing `isLaunchPage` sounds like a good milestone.

I'm investigating here more today as well.